### PR TITLE
Hw2

### PR DESCRIPTION
**Pull request by zfreiter**

### CODE REVIEW

File: hw2/01-coins.js [hunk 1]
```diff
@@ -1,7 +1,54 @@
...
+  const nickel = deleteMoney(input, 0.05);
+  input -= nickel * 0.05;
+  nickel;
+  const nickelStr = `${
+    nickel != 0 ? `, ${nickel} ${nickel > 1 ? 'nickels' : 'nickel'}` : ''
+  }`;
+
+  const penny = deleteMoney(input, 0.01);
+  const pennyStr = `${
+    penny != 0 ? `, ${penny} ${penny > 1 ? 'pennies' : 'penny'}` : ''
```
Comment (Vincentmask) [inline hw2/01-coins.js]: Looks great, I like how you assign plural to these cases

File: hw2/02-reverse.js [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+div[1].append(answer);
+
+const reverseNumber = (numberToReverse) => {
+  let rev = 0;
+  let digitToAdd = 0;
+
+  while (numberToReverse !== 0) {
+    digitToAdd = numberToReverse % 10;
+    rev = rev * 10 + digitToAdd;
+    numberToReverse = Math.floor(numberToReverse / 10);
```
Comment (Vincentmask) [inline hw2/02-reverse.js:20]: Fun way for reversing string

File: hw2/03-form.js [hunk 3]
```diff
@@ -1,3 +1,104 @@
...
+buttonDiv.className = 'col-xs-4 pt-4 pb-4';
+form.append(buttonDiv);
+
+let submitButton = document.createElement('button');
+submitButton.setAttribute('type', 'submit');
+submitButton.className = 'btn btn-primary w-50 border rounded';
+submitButton.textContent = 'Submit';
+buttonDiv.append(submitButton);
+
+let resetButton = document.createElement('button');
```
Comment (Vincentmask) [inline hw2/03-form.js]: We can modify the html code to make this program shorter

File: hw2/04-api.js [hunk 4]
```diff
@@ -1,7 +1,52 @@
...
 
-getData(url);
+let getData = (url, results) => {
+  fetch(url)
+    .then((response) => {
+      return response.json();
+    })
+    .then((data) => {
+      // Sort the data by the common name
+      data.sort((a, b) => {
```
Comment (Vincentmask) [inline hw2/04-api.js:14]: Great work for sorting here, I was too lazy to sort these so I just used Map to help

File: hw2/04-api.js [hunk 5]
```diff
@@ -1,7 +1,51 @@
...
+        const name1 = a.name.common.toUpperCase();
```
Comment (CristianTS599) [inline hw2/04-api.js:15]: Lines 14 through 26:
I Checked these from the guide.

- [Arrow Functions](https://github.com/airbnb/javascript#arrow-functions)
   - Good: your variable is wrapped in parens, your multiline function scope is wrapped in brackets so we know when it starts and ends (guide says use parens but brackets are pretty much the same) and your using the explicit return as you should for multiline.
- Variables and typing are good against the guide, good sorting.

**Review by CristianTS599 — COMMENTED** [review]
Overall very good job it was a challenge to find recommendations from the airbnb guide looking at your code. all I found was a variable that wasn't using camelCase, your variable names are descriptive, ternary's and arrow functions used appropriately, and code is well organized.

**Pull request closed without merging**